framework/src/base/FEProblem.C

```diff
@@ -1820,20 +1828,28 @@ FEProblem::addMaterial(const std::string & mat_name, const std::string & name, I
       object_name = name + "_neighbor";
       MooseSharedPointer<Material> neighbor_material = _factory.create<Material>(mat_name, object_name, current_parameters, tid);
 
+      // dirac material
+      current_parameters.set<Moose::MaterialDataType>("_material_data_type") = Moose::DIRAC_MATERIAL_DATA;
+      current_parameters.set<bool>("_dirac") = true;
+      object_name = name + "_dirac";
+      MooseSharedPointer<Material> dirac_material = _factory.create<Material>(mat_name, object_name, current_parameters, tid);
```

Comment:
Actually - I could be convinced that this is ok.

@idaholab/moose-developers what do you guys think?  Is this reasonable?

A `Material` might hold quite a lot of data in it... even if storage for its properties is never allocated...